airbyte-integrations/connectors/source-metabase/source_metabase/source.py

```diff
@@ -0,0 +1,132 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from typing import Any, Iterator, List, Mapping, MutableMapping, Tuple
+
+import requests
+from airbyte_cdk.models import AirbyteMessage, ConfiguredAirbyteCatalog
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http.auth import HttpAuthenticator
+from source_metabase.streams import Activity, Cards, Collections, Dashboards, Users
+
+API_URL = "instance_api_url"
+USERNAME = "username"
+PASSWORD = "password"
+SESSION_TOKEN = "session_token"
+
+
+class MetabaseAuth(HttpAuthenticator):
+    def __init__(self, logger: logging.Logger, config: Mapping[str, Any]):
+        self.need_session_close = False
+        self.session_token = ""
+        self.logger = logger
+        self.api_url = config[API_URL]
+        if USERNAME in config and PASSWORD in config:
+            self.username = config[USERNAME]
+            self.password = config[PASSWORD]
+        if SESSION_TOKEN in config:
+            self.session_token = config[SESSION_TOKEN]
+        elif USERNAME in config and PASSWORD in config:
+            self.session_token = self.get_new_session_token(config[USERNAME], config[PASSWORD])
+        else:
+            raise KeyError("Required parameters (username/password pair or session_token) not found")
```

Comment:
you are losing the username/password from the config when there is a session_token.

It's useful to keep them around in case the session_token is expired/invalid, so the auth can regenerate a new one instead